src/scripts/main.js

```diff
@@ -1,3 +1,269 @@
+      });
+
+      bodyRows.forEach(node => node.parentNode.removeChild(node));
+      newBodyRows4.forEach(node => tableBody.appendChild(node));
+
+      break;
+  }
+};
+
+function ascF(e, i) {
+  switch (e.target.textContent) {
+    case 'Name':
+    case 'Position':
+    case 'Office':
+      const newBodyRows1 = [...bodyRows].sort((a, b) => (
+        a.cells[i].textContent.localeCompare(b.cells[i].textContent)
+      ));
+
+      bodyRows.forEach(node => node.parentNode.removeChild(node));
+      newBodyRows1.forEach(node => tableBody.appendChild(node));
+
+      break;
+
+    case 'Age':
+    case 'Salary':
+      const newBodyRows4 = [...bodyRows].sort((a, b) => {
+        const aSalaryStr = a.cells[i].textContent.replace(/[^\d]/g, '');
+        const aSalaryValue = parseInt(aSalaryStr, 10);
+        const bSalaryStr = b.cells[i].textContent.replace(/[^\d]/g, '');
+        const bSalaryValue = parseInt(bSalaryStr, 10);
+
+        return aSalaryValue - bSalaryValue;
+      });
+
+      bodyRows.forEach(node => node.parentNode.removeChild(node));
+      newBodyRows4.forEach(node => tableBody.appendChild(node));
+
+      break;
+  }
+};
+
+document.body.insertAdjacentHTML('beforeend', `
+  <form class="new-employee-form" method="POST">
+    <label>
+      Name:
+      <input
+        name="name"
+        type="text"
+        data-qa="name"
+        required
+      >
+    </label>
+
+    <label>
+      Position:
+      <input
+        name="position"
+        type="text"
+        data-qa="position"
+        required
+      >
+    </label>
+
+    <label>
+      Office:
+      <select name="office" data-qa="office">
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+        min="0"
+        required
+      >
+    </label>
+
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+        min="0"
+        required
+      >
+    </label>
+
+    <button type="submit">Save to table</button>
+  </form>
+`);
+
+const form = document.querySelector('form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const data = new FormData(form);
+
+  const newEmployee = Object.fromEntries(data.entries());
+
+  newEmployee.salary = `$${
+    (newEmployee.salary / 1000).toFixed(3).toString().replace('.', ',')
+  }`;
+
+  const newRow = document.createElement('tr');
+
+  newRow.innerHTML = `
+    <td>${newEmployee.name}</td>
+    <td>${newEmployee.position}</td>
+    <td>${newEmployee.office}</td>
+    <td>${newEmployee.age}</td>
+    <td>${newEmployee.salary}</td>
+  `;
+
+  if (newEmployee.name.length < 4
+    && (+(newEmployee.age) < 18 || +(newEmployee.age) > 90)) {
```

Comment:
Better to move it's in variables, like: `isValidName`, `isValidAge`.  In such way you will make your code more readable.